rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/fleet_adapter.py

```diff
@@ -112,6 +112,7 @@ def initialize_fleet(config_yaml, nav_graph_path, node, use_sim_time):
         datetime.timedelta(seconds=1.0/fleet_state_update_frequency))
     # Account for battery drain
     drain_battery = fleet_config['account_for_battery_drain']
+    lane_merge_distance = fleet_config.get('lane_merge_distance', 1.0)
```

Comment:
We may want to consider decreasing the default `lane_merge_distance`, I believe it is what's causing some of the bad route generation. When filtering waypoints, for routes that include a perpendicular (or close to perpendicular) trajectory, if the robot's pose is within the `lane_merge_distance` of the corner waypoint, [this filter](https://github.com/open-rmf/rmf_demos/blob/async_behavior_fixes/rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotCommandHandle.py#L649-L652) will delete the waypoints leading to the target, resulting in the "shortcut" path. This tends to occur when the robot is provided with a new path after a negotiation, and it happens to be positioned near the corner waypoint. Tried running a few rounds of simulation with a lane merge distance of 0.1, seems to work fine so far.

Alternatively we could add `lane_merge_distance` to the full control adapters' configs if we want to keep the default at 1.0.

Reply:
Yeah you're right, 1m is unreasonable for our slotcars. They stay very close to the lanes anyway.

Reply:
https://github.com/open-rmf/rmf_demos/pull/157/commits/ef08ed498a79f0a5d63eae2be39a6ec46bf6305b